docs/manual/upgrades.md

```diff
@@ -0,0 +1,129 @@
+If the process has this condition longer than the [IgnoreMissingProcessesSeconds](../cluster_spec.md#foundationdbclusterautomationoptions) setting it will be ignored in the [Restart Phase](#restart-phase) to ensure the operator can move forward with the upgrade.
+This also means that for a short time the cluster might be running with less resources.
+
+Processes that a restarted to early can be identified with the `MissingProcesses` condition.
+Using `kubectl` to fetch the cluster status and searching for all Process Groups with this condition gives an overview of those processes restarted to early.
+
+The operator will handle those cases automatically and there is no need for manual intervention as long as not too many processes are restarted from the outside.
+In that case the best way forward is to restart all processes still running on the old version, this can either be done manually or by using the [kubectl fdb plugin](../../kubectl-fdb/Readme.md) with the following command:
+
+```bash
+kubectl fdb restart -c cluster --process-condition=IncorrectCommandLine
+```
+
+or
+
+```bash
+kubectl fdb restart -c cluster --process-condition=MissingProcesses
+```
+
+One special case here is when the majority of the coordinators are restarted and the rest of the processes are still running on the old version.
+In this case the operator is not able to handle the upgrade correctly since the returned cluster status has no information besides the coordinators that are reachable.
+During this time the FDB cluster will be unavailable.
+You have to manually restart all processes with the [kubectl fdb plugin](../../kubectl-fdb/Readme.md) with the command from above.
+
+#### Processes are not restarted during the restart Phase
+
+This issue is similar to the one above with the difference that affected processes are not restarted and they are still running on the old `fdbserver` version.
+For version incompatible upgrades that means that those processes are not reporting to the cluster.
+The operator will mark this Process Group with the `MissingProcesses` condition, since the process will be missing in the [cluster status json](https://apple.github.io/foundationdb/mr-status.html).
+If those processes are reported in the `incompatible_connections` field of the cluster status json the operator will recreate those Pods directly with the new image in the `RemoveIncompatibleProcesses` subreconciler.
+Otherwise the Pod will recreate those Pods in the [recreation of Pods phase](#recreation-of-pods-phase) with the new image.
+During that time the cluster might run with reduced capacity since not all processes are reporting to the cluster.
+
+The operator can handle those cases automatically.
+If you want to speed up this process you can restart the processes with the [kubectl fdb plugin](../../kubectl-fdb/Readme.md):
+
+```bash
+kubectl fdb restart -c cluster --process-condition=MissingProcesses
```

Comment:
So these processes are not going to be marked with "IncorrectCommandLine" condition? If not, it could be a problem as the operator/user cannot distinguish between these cases but they would need to specify a different "--process-condition" condition to resolve them.

Reply:
Also, why don't we do automated replacements in this case too (so this can be resolved without requiring human intervention)?

Reply:
They are not marked with `IncorrectCommandLine` as those processes are running on the old FDB version and therefore are not part of the cluster status json output and as they are missing they will be marked as `MissingProcesses`.

> Also, why don't we do automated replacements (that is, replace these pods) in this case too (so this can be resolved without requiring human intervention)?

If the cluster is still available the operator will recreate those Pods with the new desired spec running the new FDB version, this is done in the `RemoveIncompatibleProcesses` subreconciler as mentioned above.

Reply:
> They are not marked with IncorrectCommandLine
So the operator is marking both the processes in this category and the processes in the previous category with the same condition, but the user would need to specify different "--process-condition" option to "kubectl fdb" to resolve these scenarios? That could be a problem. Again, I think the underlying problem is that the operator is using the same condition to mark processes that are in different states.